chainer/functions/array/depth2space.py

```diff
@@ -23,23 +23,19 @@ def forward(self, inputs):
         xp = cuda.get_array_module(X)
         bsize, c, a, b = X.shape
         c //= self.r ** 2
-        X = xp.transpose(X, (0, 2, 3, 1))
-        X = xp.reshape(X, (bsize, a, b, self.r, self.r, c))
-        X = xp.transpose(X, (0, 1, 3, 2, 4, 5))
-        X = xp.reshape(X, (bsize, a * self.r, b * self.r, c))
-        X = xp.transpose(X, (0, 3, 1, 2))
+        X = xp.reshape(X, (bsize, self.r, self.r, c, a, b))
+        X = xp.transpose(X, (0, 3, 4, 1, 5, 2))
+        X = xp.reshape(X, (bsize, c, a * self.r, b * self.r))
         return X,
 
     def backward(self, indexes, grad_outputs):
         gy, = grad_outputs
```

Comment:
I think this can further improve performance by just replacing entire backward function with:

```
return chainer.functions.array.space2depth.Space2Depth(self.r).apply(grad_outputs)
```